tests/generation/test_utils.py

```diff
@@ -1467,6 +1467,17 @@ def test_past_key_values_format(self):
                         past_kv[i][1].shape, (batch_size, num_attention_heads, seq_length, per_head_embed_dim)
                     )
 
+    def test_generate_from_inputs_embeds_with_bos_token_id_is_none(self):
+        article = "Today a dragon flew over Paris."
+        model = AutoModelForCausalLM.from_pretrained("hf-internal-testing/tiny-random-gpt2").to(torch_device)
+        tokenizer = AutoTokenizer.from_pretrained("hf-internal-testing/tiny-random-gpt2")
+        input_ids = tokenizer(article, return_tensors="pt").input_ids.to(torch_device)
+        inputs_embeds = model.get_input_embeddings()(input_ids)
+
+        model.generate(inputs_embeds=inputs_embeds, max_length=20, bos_token_id=None)
+        with self.assertRaises(ValueError):
+            model.generate(max_length=20, bos_token_id=None)
```

Comment:
I think? Otherwise the error is being raised from the lack of inputs 

```suggestion
            model.generate(input_ids, max_length=20, bos_token_id=None)
```

Reply:
@amyeroberts 

I think we should not pass in `input_ids`. I tested the following code on the `main` branch and found that the generation works well (see case 3) If we pass in `input_ids` with `bos_token_id=None`.

```python
from transformers import AutoModelForCausalLM, AutoTokenizer

model = AutoModelForCausalLM.from_pretrained("hf-internal-testing/tiny-random-gpt2")
tokenizer = AutoTokenizer.from_pretrained("hf-internal-testing/tiny-random-gpt2")
article = "Today a dragon flew over Paris."
input_ids = tokenizer(article, return_tensors="pt").input_ids

# Case 1
ids = model.generate(max_length=20)[0]
print(tokenizer.decode(ids))
# �vedvedvedvedvedvedvedved Wh Wh Wh Wh Wh Wh Wh Wh Wh Wh Wh

# Case 2
ids = model.generate(input_ids, max_length=20)[0]
print(tokenizer.decode(ids))
# Today a dragon flew over Paris. fe fe fe fe

# Case 3
ids = model.generate(input_ids, max_length=20, bos_token_id=None)[0]
print(tokenizer.decode(ids))
# Today a dragon flew over Paris. fe fe fe fe

# Case 4, error
# Below code will raise a ValueError
# ids = model.generate(max_length=20, bos_token_id=None)[0]  
```